src/synapse/workload.py

```diff
@@ -348,6 +348,24 @@ def enable_federation_domain_whitelist(container: ops.Container, charm_state: Ch
         raise WorkloadError(str(exc)) from exc
 
 
+def enable_allow_public_rooms_over_federation(container: ops.Container) -> None:
+    """Change the Synapse configuration to allow public rooms in federation.
+
+    Args:
+        container: Container of the charm.
+
+    Raises:
+        WorkloadError: something went wrong enabling configuration.
+    """
+    try:
+        config = container.pull(SYNAPSE_CONFIG_PATH).read()
+        current_yaml = yaml.safe_load(config)
+        current_yaml["allow_public_rooms_over_federation"] = True
+        container.push(SYNAPSE_CONFIG_PATH, yaml.safe_dump(current_yaml))
+    except ops.pebble.PathError as exc:
+        raise WorkloadError(str(exc)) from exc
```

Comment:
What if it later gets disabled?

Reply:
If is not set, the default is False: https://ems-docs.element.io/books/element-cloud-documentation/page/import-database-and-media-dump
So the configuration is only added to the configuration file if is set to True.

Reply:
So what about if it was once set to `True` and then the user disables it after that? How would the line from the configuration end up getting removed?

Reply:
The configuration file is regenerated for every config-change event. If the user disables it, will emit a config changed event and the configuration file will be regenerated without this line.